ui/src/androidTest/java/com/hyperwallet/android/transfermethod/ui/PayPalTest.java

```diff
@@ -0,0 +1,220 @@
+                .getResourceContent("authentication_token_response.json")).mock();
+        mMockWebServer.mockResponse().withHttpResponseCode(HTTP_OK).withBody(sResourceManager
+                .getResourceContent("successful_tmc_paypal_fields_response.json")).mock();
+    }
+
+    @After
+    public void cleanup() {
+        RepositoryFactory.clearInstance();
+    }
+
+    @Before
+    public void registerIdlingResource() {
+        IdlingRegistry.getInstance().register(EspressoIdlingResource.getIdlingResource());
+    }
+
+    @After
+    public void unregisterIdlingResource() {
+        IdlingRegistry.getInstance().unregister(EspressoIdlingResource.getIdlingResource());
+    }
+
+    @Test
+    public void testAddTransferMethod_displaysElementsOnTmcResponse() {
+        mActivityTestRule.launchActivity(null);
+
+        onView(allOf(instanceOf(TextView.class), withParent(withId(R.id.toolbar)))).check(
+                matches(withText(R.string.paypal_account)));
+
+        onView(withId(R.id.email)).check(matches(isDisplayed()));
+        onView(withId(R.id.emailLabel)).check(matches(withHint("Email")));
+
+        onView(withId(R.id.add_transfer_method_button)).perform(nestedScrollTo()).check(
+                matches(withText(R.string.button_create_transfer_method)));
+    }
+
+    @Test
+    public void testAddTransferMethod_displaysFeeElementsOnTmcResponse() {
+        mActivityTestRule.launchActivity(null);
+
+        onView(withId(R.id.add_transfer_method_fee_label)).check(
+                matches(withText(R.string.add_transfer_method_fee_label)));
+        onView(withId(R.id.add_transfer_method_processing_label)).check(
+                matches(withText(R.string.add_transfer_method_processing_time_label)));
+        onView(withId(R.id.add_transfer_method_fee_value)).check(matches(withText("USD 0.25")));
+        onView(withId(R.id.add_transfer_method_processing_time_value)).check(matches(withText("IMMEDIATE")));
+    }
+
+    @Test
+    public void testAddTransferMethod_returnsTokenOnPaypalAccountCreation() throws InterruptedException {
+        mMockWebServer.mockResponse().withHttpResponseCode(HTTP_CREATED).withBody(sResourceManager
+                .getResourceContent("paypal_response.json")).mock();
+
+        mActivityTestRule.launchActivity(null);
+
+        final CountDownLatch gate = new CountDownLatch(1);
+        final BroadcastReceiver br = new BroadcastReceiver() {
+            @Override
+            public void onReceive(Context context, Intent intent) {
+                gate.countDown();
+
+                HyperwalletTransferMethod transferMethod = intent.getParcelableExtra(
+                        "hyperwallet-local-broadcast-payload");
+                assertThat("Bank Account Id is incorrect", transferMethod.getField(
+                        HyperwalletTransferMethod.TransferMethodFields.EMAIL), is("sunshine.carreiro@felicity.com"));
```

Comment:
what email domain is that? maybe just something with @hyperwallet.com
please update other places as well

Reply:
updated